### ISSUE TITLE
Use well-known (Docker | OCI) Labels to expose metadata such as source URL

### ISSUE THREAD

**Issue by Oro** [k8ssandra/k8ssandra#474]
**Is your feature request related to a problem? Please describe.**
Docker (as well as all OCI runtimes) use well-known labels - https://github.com/opencontainers/image-spec/blob/master/annotations.md - to populate some Metadata inside e.g. the Registries.
This includes things like build-date, but more importantly links to the repository, the documentation and the licence.

**Describe the solution you'd like**
Include the well-known labels in all Dockerfiles that are built in k8ssandra//datastax (cassandra-mgmtapi, cass-operator, medusa-operator, medusa, reaper-operator and create-keyspace from what I can tell)
This would be something like the following in each of the Dockerfiles
```Dockerfile
LABEL org.opencontainers.image.created="{{buildDate}}" \
      org.opencontainers.image.documentation="https://k8ssandra.io/docs/reference/k8ssandra/" \
      ...
```

**Describe alternatives you've considered**
n/a

**Additional context**
This is just a nice-to-have, knowing the licence or where the sources/docs of a particular Image are located with a glance at the metadata would be neat.

**Comment by bradfordcp** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/K8ssandra-Grooming-623b197f1fcd4800110540ef/issues/k8ssandra/k8ssandra/474?planning-poker) with ZenHub @burmanm @jsanda @Miles-Garnsey

**Comment by jsanda** [comment]
This should be broken out into multiple tickets, maybe by image, but certainly by repo.

**Comment by bradfordcp** [comment]
Do we want to use this as an epic across those tickets?

**Comment by jsanda** [comment]
Yes, let's create an epic. I think we should create it through ZenHub though.
